src/Tmds.Ssh/SftpClient.File.cs

```diff
@@ -0,0 +1,161 @@
+        internal async ValueTask<bool> SendCloseHandleAsync(byte[] handle)
+        {
+            using var packet = CreateCloseMessage(handle);
+            var operation = new CloseHandleOperation();
+
+            await SendRequestAsync(packet.Move(), operation);
+
+            return await operation.Task;
+
+            Packet CreateCloseMessage(byte[] handle)
+            {
+                using var packet = _context.RentPacket();
+                var writer = packet.GetWriter();
+                writer.Reserve(DataHeaderLength);
+
+                /*
+                    SSH_FXP_CLOSE
+                    uint32        id
+                    string        handle
+                */
+
+                writer.WriteSftpPacketType(SftpPacketType.SSH_FXP_CLOSE);
+                writer.WriteUInt32(0);
+                writer.WriteString(handle);
+                return packet.Move();
+            }
+        }
+    }
+
+    sealed class OpenFileOperation : SftpOperation
+    {
+        private TaskCompletionSource<SftpFile> _tcs = new TaskCompletionSource<SftpFile>();
+
+        public override ValueTask HandleResponse(SftpPacketType type, ReadOnlySequence<byte> fields, SftpClient client)
+        {
+            if (type == SftpPacketType.SSH_FXP_STATUS)
+            {
+                _tcs.SetException(CreateExceptionForStatus(fields));
+            }
+            else if (type == SftpPacketType.SSH_FXP_HANDLE)
+            {
+                var handle = ParseHandleFields(fields);
+                _tcs.SetResult(new SftpFile(handle, client));
+            }
+            else
+            {
+                _tcs.SetException(CreateExceptionForUnexpectedType(type));
+            }
+
+            return default;
+        }
+
+        public Task<SftpFile> Task => _tcs.Task;
+
+        static byte[] ParseHandleFields(ReadOnlySequence<byte> fields)
+        {
+            /*
+                byte   SSH_FXP_HANDLE
+                uint32 request-id
+
+                string handle
+            */
+            var reader = new SequenceReader(fields);
+            byte[] handle = reader.ReadStringAsBytes().ToArray();
+            return handle;
+        }
+    }
+
+    // This can be used for any handle, so also for directories
+    sealed class CloseHandleOperation : SftpOperation
+    {
+        private TaskCompletionSource<bool> _tcs = new TaskCompletionSource<bool>();
+
+        // TODO can return status and on some systems even close can fail
+        public override ValueTask HandleResponse(SftpPacketType type, ReadOnlySequence<byte> fields, SftpClient client)
+        {
+            if (type != SftpPacketType.SSH_FXP_STATUS)
+            {
+                _tcs.SetException(CreateExceptionForStatus(fields));
+            }
+            else
+            {
+                // if (SSH_FXP_STATUS == success) else.....
+                _tcs.SetResult(true); // TODO parsing of SSH_FXP_STATUS
```

Comment:
This can use the same method as `CreateExceptionForStatus` to parse status fields.
The method should be in `SftpOperation` and have a signature that looks like:

```
protected static (SftpErrorCode errorCode, string? errorMessage) ParseStatusFields(ReadOnlySequence<byte> fields)
```